codesign/lib/src/file_codesign_visitor.dart

```diff
@@ -164,4 +184,51 @@ class FileCodesignVisitor {
       processManager: processManager,
     );
   }
+
+  /// Visit and codesign a binary with / without entitlement.
+  ///
+  /// At this stage, the virtual [entitlementCurrentPath] accumulated through the recursive visit, is compared
+  /// against the paths extracted from [fileWithEntitlements], to help determine if this file should be signed
+  /// with entitlements.
+  Future<void> visitBinaryFile({required File binaryFile, required String entitlementParentPath}) async {
+    if (io.FileSystemEntity.isLinkSync(binaryFile.absolute.path)) {
+      return;
+    }
+
+    final String currentFileName = binaryFile.absolute.path.split('/').last;
```

Comment:
```suggestion
    final String currentFileName = binaryFile.basename;
```

Reply:
Very true, I should have caught them all.